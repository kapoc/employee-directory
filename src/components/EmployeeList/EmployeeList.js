import React, { useState, useEffect } from 'react';
import EmployeeListItem from '../EmployeeListItem/EmployeeListItem';
import store from '../../store/store';

function EmployeeList() {
  const [employees, setEmployees] = useState(store.getState().employees);

  useEffect(() => {
    const unsubscribe = store.subscribe(() => {
      setEmployees(store.getState().employees);
    });

    return () => {
      unsubscribe();
    }
  }, []);

  return (
    <div>
      <ul className='list-group'>
        {employees.map((employee) => {
          return <EmployeeListItem key={employee.id} {...employee}/>
        })}
      </ul>
    </div>
  );
}

export default EmployeeList;
