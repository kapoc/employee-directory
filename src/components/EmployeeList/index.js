import React from 'react';
import EmployeeListItem from '../EmployeeListItem';
import store from '../../store/store';

class EmployeeList extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      employees: store.getState().employees
    }

    store.subscribe(() => {
      this.setState({ employees: store.getState().employees })
    });
  }

  render(){
    return (
      <div>
        <ul className='list-group'>
          {this.state.employees.map((employee) => {
            return <EmployeeListItem key={employee.id} {...employee}/>
          })}
        </ul>
      </div>
    );
  }
}

export default EmployeeList;
