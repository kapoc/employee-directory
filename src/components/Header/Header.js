import React from 'react';

function Header({title}){
  return (
    <div className='header'>
      <h4 className='header-title'>
        {title}
      </h4>
    </div>
  )
}

export default Header;
