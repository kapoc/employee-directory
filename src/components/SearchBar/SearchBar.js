import React from 'react';
import store from '../../store/store';
import { filterEmployees } from '../../actions/actions';

function SearchBar() {
  function onChange(event) {
    store.dispatch(filterEmployees(event.target.value));
  }

  return (
    <div className="form-group" id='searchbar-wrapper'>
      <input type="text" className="form-control" placeholder="Search employees" onChange={onChange}/>
    </div>
  );
}

export default SearchBar;
