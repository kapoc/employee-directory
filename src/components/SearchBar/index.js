import React from 'react';
import store from '../../store/store';
import { setSearchQuery } from '../../actions/actions';

class SearchBar extends React.Component {
  constructor(props){
    super(props);
  }

  onChange = (event) => {
    store.dispatch(setSearchQuery(event.target.value));
  }

  render(){
    return (
      <div className="form-group">
        <input type="text" className="form-control" placeholder="Search employees" onChange={this.onChange}/>
      </div>
    );
  }
}

export default SearchBar;
