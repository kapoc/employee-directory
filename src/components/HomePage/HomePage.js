import React from 'react';
import Header from '../Header/Header';
import SearchBar from '../SearchBar/SearchBar';
import EmployeeList from '../EmployeeList/EmployeeList';

function HomePage() {
  const title = 'Employee Directory';

  return (
    <div className='card'>
      <div className='card-body scrollbar-pretty'>
        <Header key={title} title={title} />

        <SearchBar/>

        <EmployeeList/>
      </div>
    </div>
  );
}

export default HomePage;
