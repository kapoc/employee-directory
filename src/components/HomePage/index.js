import React from 'react';
import Header from '../Header';
import SearchBar from '../SearchBar';
import EmployeeList from '../EmployeeList';

class HomePage extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      title: 'Employee Directory',
    }
  }

  render(){
    return (
      <div className='card'>
        <div className='card-body scrollbar-pretty'>
          <Header key={this.state.title} title={this.state.title} />

          <SearchBar/>

          <EmployeeList/>
        </div>
      </div>
    );
  }
}

export default HomePage;
