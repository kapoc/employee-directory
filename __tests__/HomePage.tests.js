import React from 'react';
import HomePage from '../src/components/HomePage/HomePage';
import Header from '../src/components/Header/Header';
import SearchBar from '../src/components/SearchBar/SearchBar';
import EmployeeList from '../src/components/EmployeeList/EmployeeList';
import renderer from 'react-test-renderer';

describe('HomePage tests', () => {
  it('Ensure homepage has it\'s structure set correctly ', () => {
    const component = renderer.create(
      <HomePage />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();


    let testInstance = component.root;
    expect(testInstance.findByType(Header)).not.toBe(null);
    expect(testInstance.findByType(SearchBar)).not.toBe(null);
    expect(testInstance.findByType(EmployeeList)).not.toBe(null);
  });
});
