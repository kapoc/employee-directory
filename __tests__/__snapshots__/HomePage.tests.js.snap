// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage tests Ensure homepage has it's structure set correctly  1`] = `
<div
  className="card"
>
  <div
    className="card-body scrollbar-pretty"
  >
    <div
      className="header"
    >
      <h4
        className="header-title"
      >
        Employee Directory
      </h4>
    </div>
    <div
      className="form-group"
      id="searchbar-wrapper"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Search employees"
        type="text"
      />
    </div>
    <div>
      <ul
        className="list-group"
      />
    </div>
  </div>
</div>
`;
